src/handlers/askingArticleSubmissionReason.js

```diff
@@ -1,46 +1,113 @@
+  if (event.input !== 'n' && event.input !== REASON_PLACEHOLDER) {
     const reason = event.input;
+    const text =
+      `以下是您所填寫的理由：\n「\n${reason}\n」\n` +
+      '我們即將把此訊息與您填寫的理由送至資料庫。' +
+      '若您送出的訊息或理由意味不明、造成闢謠編輯的困擾，可能會影響到您未來送出文章的權利。' +
+      '\n' +
+      '若要確認請輸入「y」、若要放棄請輸入「n」、若要重新填寫理由請輸入「r」';
 
     replies = [
       {
-        type: 'text',
-        text: `以下是您所填寫的理由：\n「\n${reason}\n」`,
-      },
-      {
-        type: 'text',
-        text:
-          '我們即將把此訊息與您填寫的理由送至資料庫。若您送出的訊息或理由意味不明、' +
-          '造成闢謠編輯的困擾，可能會影響到您未來送出文章的權利。',
-      },
-      {
-        type: 'template',
-        altText:
-          '若要確認請輸入「y」、若要放棄請輸入「n」、若要重新填寫理由請輸入「r」',
-        template: {
-          type: 'buttons',
-          text: '請確認',
-          actions: [
-            createPostbackAction('明白我要送出', 'y', issuedAt),
-            createPostbackAction('重寫送出的理由', 'r', issuedAt),
-            createPostbackAction('放棄送出', 'n', issuedAt),
-          ],
+        type: 'flex',
+        altText: text,
+        contents: {
+          type: 'bubble',
+          styles: {
+            footer: {
+              separator: true,
+            },
+          },
+          body: {
+            type: 'box',
+            layout: 'vertical',
+            contents: [
+              {
+                type: 'text',
+                text: '把訊息送進資料庫',
+                weight: 'bold',
+                color: '#1DB446',
+                size: 'sm',
+              },
+              {
+                type: 'separator',
+                margin: 'xxl',
+              },
+              {
+                type: 'text',
+                text: '訊息',
+                weight: 'bold',
+                size: 'xl',
+                margin: 'xl',
+              },
+              {
+                type: 'text',
+                text: data.searchedText,
+                size: 'xs',
+              },
+              {
+                type: 'separator',
+                margin: 'xxl',
+              },
+              {
+                type: 'text',
+                text: '覺得是謠言的理由',
+                weight: 'bold',
+                size: 'xl',
+                margin: 'xl',
+              },
+              {
+                type: 'text',
+                text: reason,
+                size: 'xxs',
+              },
+              {
+                type: 'separator',
+                margin: 'xxl',
+              },
+              {
+                type: 'text',
+                text:
+                  '若您送出的訊息或理由意味不明、造成闢謠編輯的困擾，未來您將無法送出文章。',
+                color: '#ff0000',
+                wrap: true,
+                margin: 'xxl',
+              },
+              {
+                type: 'button',
+                margin: 'xl',
+                style: 'primary',
+                action: createPostbackAction('放棄送出', 'n', issuedAt),
+              },
+              {
+                type: 'button',
+                action: createPostbackAction('重寫送出的理由', 'r', issuedAt),
```

Comment:
Currently clicking "重寫送出的理由" would be a bit weird:

![screenshot_20180807-012341](https://user-images.githubusercontent.com/108608/43731302-a5c08fe6-99e0-11e8-94af-6d3e8406451e.png)

I would suggest the LINE bot should ask "送出訊息到公開資料庫？" again and provide "放棄送出" and "傳理由給我們" after clicking “重寫送出的理由” as well (i.e. reuse `createAskArticleSubmissionReply`), so that the user knows what to rewrite.

Reply:
If we reuse `createAskArticleSubmissionReply`, users may find it confusing since they don't know if they need to click "傳理由給我們" again. “重寫送出的理由” sounds like that I can enter my revised reason directly without any extra action.
However, if we reuse this but provide "放棄送出" only, users may still feel confused due to inconsistency between these two "almost identical" messages, say, "現在只剩下放棄送出，是我只能放棄的意思嗎？".

Reply:
I think the reuse is ok because
1. I think it is acceptable if user spends an extra click on  "⌨️ 傳理由給我們". At least it provides obvious call-to-action (green button) instead of just a text.
2. We cannot provide "放棄送出" only. Like you said, providing this option solely is quite confusing. That's why my proposal suggests providing both buttons.

If 1 is not acceptable, how about changing "重寫送出的理由" to something like "回上一步" so that the call-to-action makes more sense.

Reply:
Meeting conclusion:

Flex message is a 簡便解法, let's keep it simple. No changes needed.

Reason:

此步驟會有問題，大部分是使用者「錯傳新訊息進來」。但若要根除此問題，就應該用更複雜的 LIFF。

現階段應該只是先試試看把送出訊息流程用 flex message 呈現之後，是否可以降低誤傳到可以接受的程度，因此不應該把流程搞得太複雜，故保留「好的，請重新填寫理由」為佳。